spark/src/main/java/org/opensearch/sql/spark/flint/operation/FlintIndexOp.java

```diff
@@ -145,11 +148,18 @@ public void cancelStreamingJob(
     String jobId = flintIndexStateModel.getJobId();
     try {
       emrServerlessClient.cancelJobRun(
-          flintIndexStateModel.getApplicationId(), flintIndexStateModel.getJobId());
-    } catch (IllegalArgumentException e) {
-      // handle job does not exist case.
+          flintIndexStateModel.getApplicationId(), flintIndexStateModel.getJobId(), true);
+    } catch (ResourceNotFoundException e) {
+      // JobId Not Found Exception.
       LOG.error(e);
       return;
+    } catch (ValidationException e) {
```

Comment:
This is for scenario where the current Job is already in terminal state, where we want to ignore and proceed with the operation.

Reply:
Currently we are capturing IllegalArgumentException and returning false. 
@penghuo  @dai-chen  Need some inputs on what exceptions should we ignore and propagate others.
/ handle job does not exist case. [@penghuo  I saw this comment. You have any specific case in mind here.]

Reply:
current change make sense to me, plugin can handle validation exception
but any other exception from EMR-S plugin can not handle. throw exception to client looks good.